src/Symfony/Component/HttpClient/Internal/AmpClientState.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\HttpClient\Internal;
+
+use Amp\Http\Client\Connection\DefaultConnectionFactory;
+use Amp\Http\Client\Connection\LimitedConnectionPool;
+use Amp\Http\Client\Connection\UnlimitedConnectionPool;
+use Amp\Http\Client\HttpClientBuilder;
+use Amp\Socket\DnsConnector;
+use Amp\Socket\StaticConnector;
+use Amp\Sync\LocalKeyedSemaphore;
+
+/**
+ * Internal representation of the Amp client's state.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ *
+ * @internal
+ */
+final class AmpClientState extends ClientState
+{
+    /** @var HttpClientBuilder[] */
+    private $clients = [];
+    private $maxHostConnections = 0;
+    private $builder;
+
+    public function __construct(?HttpClientBuilder $builder, int $maxHostConnections)
+    {
+        $this->builder = $builder ?? (new HttpClientBuilder())->retry(0);
```

Comment:
What's the reason to disable retries?

Reply:
disabling retries for consistency with the other implementations

Reply:
What's considered a retry for other implementations? In our HTTP client these are e.g. a closing HTTP/2 connection.

Reply:
CurlHttpClient has a retry to workaround errors raised by curl in HTTP/2 mode (related to bugs in the implementation apparently, fixed in recent versions): before the body arrives and when an HTTP/2 specific error happens, we retry once in h2, then in h1.1.
I'm going to keep the retry if you think it improves the robustness of the client.

Reply:
If there's retry logic in other places, it might be fine to disable the retries here. If retries are enabled, only requests guaranteed to be unprocessed by the server or idempotent and a socket error happened before receiving the response headers are going to be retried.